src/codecs/TypingNotification.ts

```diff
@@ -0,0 +1,49 @@
+import { ContentTypeId, ContentCodec, EncodedContent } from '../MessageContent'
+
+// xmtp.org/typingNotification
+//
+// This content type is used for typing notifications
+export const ContentTypeTypingNotification = new ContentTypeId({
+  authorityId: 'xmtp.com',
+  typeId: 'typingNotification',
+  versionMajor: 1,
+  versionMinor: 0,
+})
+
+export type TypingNotification = {
+  timestamp: Date
+  typerAddress: string
+  isFinished: boolean
+}
+
+export class TypingNotificationCodec
+  implements ContentCodec<TypingNotification>
+{
+  get contentType(): ContentTypeId {
+    return ContentTypeTypingNotification
+  }
+
+  encode(content: TypingNotification): EncodedContent {
+    return {
+      type: ContentTypeTypingNotification,
+      parameters: {
+        timestamp: content.timestamp.toISOString(),
+        typerAddress: content.typerAddress,
+        isFinished: content.isFinished ? 'true' : 'false',
```

Comment:
>  I wonder if regularly sending a keep-alive payload with a TTL for the duration that typing is happening would be more appropriate

That's how I've had it implemented, but I was finding there were some races that would lead to the typing indicator showing up after a message had come in. This field was a way to just tell listeners "we're done here".